File: src/graph_manager/models/graph.py
from pydantic import BaseModel
from typing import List, Optional, Literal

class GraphNode(BaseModel):
    id: str
    label: str
    type: Literal["job", "table"]
    status: Literal["success", "failure", "pending"] = "pending"
    enabled: bool = True
    meta: Optional[dict] = None

class GraphEdge(BaseModel):
    id: str
    source: str
    target: str
    label: Optional[str] = None

class GraphData(BaseModel):
    nodes: List[GraphNode]
    edges: List[GraphEdge]

File: apps/admin_console/lineage/src/types/graph.ts
export interface GraphNode {
  id: string;
  type: string;
  name: string;
  full_name?: string;
  status?: string;
  label?: string;
  [key: string]: any;
}

export interface GraphEdge {
  source: string;
  target: string;
  type?: string;
}

export interface GraphState {
  nodes: GraphNode[];
  edges: GraphEdge[];
}

export type LayoutOrientation = "LR" | "TB";
export type SelectionType = "job" | "table";

export interface Selection {
  type: SelectionType;
  id?: string;
  jobId?: string;
  tableName?: string;
}

export type SelectHandler = (selection: Selection | null) => void;

export interface ContextMenuState {
  x: number;
  y: number;
  node: GraphNode;
}
